stories/utils/SettingsPanel.tsx

```diff
@@ -19,6 +22,81 @@ interface ISettingsPanelConfig {
   title: string;
 }
 
+export const autoSettingsPanel = () =>
+  plugin({
+    handler: (section, storyConfig) => {
+      const componentName = storyConfig.storyName;
+
+      const variables = storyConfig.metadata.stylable.overridableVars;
+
+      const numbers = Object.values(variables)
+        .filter(
+          (variableInfo: IVariableManifest) =>
+            variableInfo.type === VariableType.number,
+        )
+        .map((variableInfo: IVariableManifest) => ({
+          label: variableInfo.name,
+          wixParam: `${componentName}-${variableInfo.name}`,
+          defaultNumber: Number(variableInfo.defaultValue),
+          unit: variableInfo.unit || '',
+          min: Number(variableInfo.min) || 0,
+          max: Number(variableInfo.max) || 100,
+        }));
+
+      const fonts = Object.values(variables)
+        .filter(
+          (variableInfo: IVariableManifest) =>
+            variableInfo.type === VariableType.font,
+        )
+        .map((variableInfo: IVariableManifest) => ({
+          label: variableInfo.name,
+          wixParam: `${componentName}-${variableInfo.name}`,
+          defaultFont: 'arial',
+        }));
+
+      const colors = Object.values(variables)
+        .filter(
+          (variableInfo: IVariableManifest) =>
+            variableInfo.type === VariableType.color,
+        )
+        .map((variableInfo: IVariableManifest) => {
+          const colorMatch = variableInfo.defaultValue.match(/color-[0-9]+/);
+          const defaultColor = colorMatch ? colorMatch[0] : 'color-1';
+
+          return {
+            label: variableInfo.name,
+            wixParam: `${componentName}-${variableInfo.name}`,
+            defaultColor,
+          };
+        });
+
+      const styleExample = require(`!raw-loader!../../src/connected-components/${componentName}/${componentName}.example.st.css`);
```

Comment:
oof this can be heavy, if i recall correctly having dynamic path in `require` moves the whole file system into the bundle (might be even with node_modules). 

this would increase storybook load time quite a lot. Please double check

on a side note, is this really the way to go? is it guaranteed that all `connected-components` have `*.example.st.css`

Reply:
That is not exactly correct as far as I know. Dynamic require makes WebPack perform static analysis on code and include whole folder (not whole file system) into the bundle. Currently this folder contains 4 files which is not a lot. Even it it had more - all of those files would be needed in bundle.

Having said that, idea is to get rid of these files (if I am lucky enough - today, if not - maybe will need to leave it for someone else). Initial idea was to generate wrappers which connect Stylable to settings. I have a working generation script in a separate branch (not meant to be merged anymore).

Afterwards we spoke with Jonathan and decided to leave these files sort-of hard-coded for now and aim to generate wrappers in memory instead of generating files. So if that succeeds, these files will be just removed.

Reply:
hmm i'm not sure about it. i remember having lots of tweaks with things like `require.resolve` and weird regex'es.

Also, `storyName` is not necessarily same as component name.
I didn't delve into whole code in this PR so can't say if this approach is good or not. It just feels very fragile

Reply:
This is sort-of temporary solution with intention to replace it with dynamic wrapper generation. I am looking into this right now but not sure if I will be able to complete this part due to Stylable generation challenges.

Worst case, will try to tidy this place up a bit and add a fat todo :)

Reply:
> Worst case, will try to tidy this place up a bit and add a fat todo :)

for sure, don't sweat it, we'll look into it later on.

But we'll look into it only if there's a fat todo :D